integration-tests/firstdata_test.go

```diff
@@ -0,0 +1,217 @@
+// +build integration
+
+package test
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+	"github.com/BoltApp/sleet/gateways/firstdata"
+	sleet_testing "github.com/BoltApp/sleet/testing"
+
+	"github.com/google/uuid"
+)
+
+var (
+	apiKey    = getEnv("FIRSTDATA_API_KEY")
+	apiSecret = getEnv("FIRSTDATA_API_SECRET")
+)
+
+func TestFirstdataAuthCapture(t *testing.T) {
+	client := firstdata.NewClient(common.Sandbox, firstdata.Credentials{apiKey, apiSecret})
+
+	authRequest := sleet_testing.BaseAuthorizationRequest()
+
+	*authRequest.ClientTransactionReference = uuid.New().String()
+	authRequest.Amount.Amount = 100
+
+	auth, err := client.Authorize(authRequest)
+
+	if err != nil {
+		t.Fatalf("Got runtime error while running authorize %q", err)
+	}
+
+	if !auth.Success {
+		t.Fatalf("Auth request should have been successful : Error Code %q", auth.ErrorCode)
+	}
+
+	reqId := uuid.New().String()
+
+	capResp, err := client.Capture(&sleet.CaptureRequest{
+		Amount:                     &authRequest.Amount,
+		TransactionReference:       auth.TransactionReference,
+		ClientTransactionReference: &reqId,
+	})
+
+	if err != nil {
+		t.Fatalf("Got runtime error while running capture %q", err)
+	}
+
+	if capResp.ErrorCode != nil {
+		t.Errorf("Expected No Error Code: received: %s", *capResp.ErrorCode)
+	}
+
+	inq, err := transactionInquiry(uuid.New().String(), capResp.TransactionReference)
```

Comment:
why dont we use the captureRequest UUID here?

Reply:
The first uuid param here is supposed to be newly generated for every request, used for their internal tracking. The second parameter is the reference to the transaction we are querying.